src/canvas.tsx

```diff
@@ -633,9 +633,9 @@ export const useCanvas = (props: UseCanvasProps): DomEventHandlers => {
       defaultScene,
       state
     )
-  }, [ready, children, sharedState.current])
+  }, [ready, children, sharedState.current]) // React Hook React.useLayoutEffect has missing dependencies: 'Canvas' and 'defaultScene'. Either include them or remove the dependency array. Mutable values like 'sharedState.current' aren't valid dependencies because mutating them doesn't re-render the component.
```

Comment:
here is possible missing hook dependency. please clarify intention

Reply:
you can add canvas and defaultscene, sharedState.current is specific, it can't just be sharedState

Reply:
yes, I understand that ref object can't be used to trigger hook, but changing ref.current does not trigger re-render by itself.

Reply:
sharedstate changes when [size, defaultCam] change, then it triggers a render, but these two can also be added to the render part, just a little more fragile because if i decide to adapt when sharedstate changes, i now must adapt it down there, too. i would leave it for now. this will all go away with zustand.

Reply:
<img width="422" alt="Screen Shot 2020-11-22 at 3 38 07 PM" src="https://user-images.githubusercontent.com/2255993/99899030-c1bed800-2cd8-11eb-9294-3e459cf84ce4.png">

Reply:
i know but it's obviously wrong. if we remove that r3f doesnt update useThree subscribers.

Reply:
I've commented out a warning, and added comment.

Reply:
great. im glad all these special cases will be gone in v6 - i cant see that they are a little weird